exception
    不断向外寻找，不匹配组合总就是终止
    析构函数里面的异常应该自己处理，因为异常的时候会销毁对象调用系够，这个时候再异常
        就会出错，构造的一场要自己处理！！不然对象不完整
